### ISSUE TITLE
[Torlock] many duplicates

### ISSUE THREAD

**Issue by annomatik** [Jackett/Jackett#6256]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue. Duplicating issues makes it more difficult for everyone to follow. Your issue may have already been solved in the past as well.

Provide a description of the feature request or bug, the more details the better.

If you are experiencing an issue with a tracker, then:
1. Use your browser to check you can access the site directly, and if a login is required, check you can login and that you do not have any outstanding account issues.
2. If you haven't already, try upgrading to the latest version of Jackett, your issue may have already been resolved.
3. If it is still not working for you, then a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting

(At the bottom of your Jackett Dashboard page) **Jackett version**:  0.11.802 on RPi 3B+ / Raspbian Stretch
(if you installed using Jackett.Binaries.Mono.tar.gz) **Mono version**: using the ARM 32 version.

-------------------------------

Hi guys, I noticed today, that with Torlock, I get a lot of duplicate search results. Please see here:

![image](https://user-images.githubusercontent.com/28311452/67634840-f9cece00-f8c0-11e9-83f3-b8b02c63596c.png)

Steps to reproduce:

1) add Torlock to Configured Indexers
2) Click "Manual Search"
3) search for stuff, e.g. "Black Jesus".

![image](https://user-images.githubusercontent.com/28311452/67634851-0f43f800-f8c1-11e9-857a-0de861ec9450.png)

**Comment by garfield69** [comment]
yes. there appears to be a bug (or missing implementation) of the path category filtering in the cardigann processor, which means that the torlock indexer was posting 11 search requests (one for every category) instead of a single search request as intended.
Since I cannot fix the cardigann processor (it requires a c# dev volunteer to debug/code a fix), I have  bypassed the issue for now, by altering the torlock indexer to just search using the **all** default search query.
try out jackett pre-release 0.12.882

**Comment by annomatik** [comment]
Cool, Thanks! Will have a look.

**Comment by ngosang** [comment]
@garfield69 How I can help to fix this? Could you give me more datails/examples?

**Comment by garfield69** [comment]
set up the following test environment:
1. create `test6256.yml` and save to your jackett definitions folder
```yml
---
  site: test6256
  name: test6256
  description: "testing path category mappings"
  language: en-us
  type: public
  encoding: UTF-8
  links:
    - https://www.torlock.com/
    - https://www.torlock2.com/
  legacylinks:
    - https://torlock.com/

  caps:
    categorymappings:
      - {id: TELEVISION, cat: TV, desc: "TV Shows"}
      - {id: MOVIES, cat: Movies, desc: "Movies"}
      - {id: MUSIC, cat: Audio, desc: "Music"}
      - {id: GAMES, cat: PC/Games, desc: "Games"}
      - {id: SOFTWARE, cat: PC, desc: "Software"}
      - {id: ANIME, cat: TV/Anime, desc: "Anime"}
      - {id: EBOOKS, cat: Books/Ebook, desc: "Books"}
      - {id: OTHER, cat: Other, desc: "Other"}
      - {id: ADULT, cat: XXX, desc: "Adult"}
      - {id: AUDIOBOOK, cat: Audio/Audiobook, desc: "Audiobook"}
      - {id: IMAGES, cat: Other/Misc, desc: "Images"}
      
    modes:
      search: [q]
      tv-search: [q, season, ep]
      movie-search: [q]
  
  settings:
    - name: sort
      type: select
      label: Sort requested from site (only works for search with keywords)
      default: "added"
      options:
        "added": "created"
        "seeds": "seeders"
        "size": "size"
    - name: type
      type: select
      label: Order requested from site
      default: "desc"
      options:
        "desc": "desc"
        "asc": "asc"
  
  search:
    paths:
#      - path: "{{ if .Keywords }}/all/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/fresh.html{{end}}"
      - path: "{{ if .Keywords }}/television/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/television/1/added/desc.html{{end}}"
        categorymappings: [TELEVISION]
      - path: "{{ if .Keywords }}/movies/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/movies/1/added/desc.html{{end}}"
        categorymappings: [MOVIES]
      - path: "{{ if .Keywords }}/music/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/music/1/added/desc.html{{end}}"
        categorymappings: [MUSIC]
      - path: "{{ if .Keywords }}/games/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/games/1/added/desc.html{{end}}"
        categorymappings: [GAMES]
      - path: "{{ if .Keywords }}/software/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/software/1/added/desc.html{{end}}"
        categorymappings: [SOFTWARE]
      - path: "{{ if .Keywords }}/anime/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/anime/1/added/desc.html{{end}}"
        categorymappings: [ANIME]
      - path: "{{ if .Keywords }}/ebooks/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/ebooks/1/added/desc.html{{end}}"
        categorymappings: [EBOOKS]
      - path: "{{ if .Keywords }}/unknown/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/unknown.html{{end}}" # sorting is broken for unknown
        categorymappings: [OTHER]
      - path: "{{ if .Keywords }}/adult/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/adult/1/added/desc.html{{end}}"
        categorymappings: [ADULT]
      - path: "{{ if .Keywords }}/audiobooks/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/audiobooks/1/added/desc.html{{end}}"
        categorymappings: [AUDIOBOOK]
      - path: "{{ if .Keywords }}/images/torrents/{{ .Keywords }}.html?sort={{ .Config.sort }}&order={{ .Config.type }}{{else}}/images/1/added/desc.html{{end}}"
        categorymappings: [IMAGES]
    keywordsfilters:
      - name: tolower
      - name: re_replace
        args: ["[^a-zA-Z0-9]+", "-"]

    rows:
      selector: table > tbody > tr:has(td:has(div:has(a[href^="/torrent/"])))
      filters:
        - name: andmatch

    fields:
      category:
        selector: span[class^="tv"]
        attribute: class
        case:
          span.tv0: OTHER
          span.tv1: MOVIES
          span.tv2: MUSIC
          span.tv3: TELEVISION
          span.tv4: GAMES
          span.tv5: SOFTWARE
          span.tv6: ANIME
          span.tv7: ADULT
          span.tv8: EBOOKS
          span.tv9: IMAGES
          span.tv12: AUDIOBOOK
          "*": "" # some torrents have invalid categories
      title:
        selector: td:nth-child(1) > div > a
      details:
        selector: td:nth-child(1) > div > a[href^="/torrent/"]
        attribute: href
      download:
        selector: td:nth-child(1) > div > a[href^="/torrent/"]
        attribute: href
        filters:
        - name: replace
          args: ["/torrent/", "/tor/"]
        - name: regexp
          args: (^/tor/\d*)
        - name: append
          args: ".torrent"
      date:
        selector: td:nth-child(2):contains("/")
        optional: true
        filters: 
          - name: dateparse
            args: "1/2/2006"
      date:
        selector: td:nth-child(2):contains("Today"), td:nth-child(2):contains("Yesterday")
        optional: true
        filters: 
          - name: fuzzytime
      date:
        selector: td:nth-child(2):not(:contains("Today")):not(:contains("Yesterday")):not(:contains("/"))
        optional: true
        filters: 
          - name: re_replace
            args: ["(min|mins)", "minutes"]
          - name: timeago
      size:
        selector: td:nth-child(3)
      seeders:
        selector: td:nth-child(4)
      leechers:
        selector: td:nth-child(5)
      downloadvolumefactor:
        text: 0
      uploadvolumefactor:
        text: 1
# Engine n/a
```
2. restart jackett service
3. enable enhanced logging
4. add test6256 indexer to your active dashboard
4. search with indexer test6256 for `black jesus` with the category filter set to `5000 TV`
5. examine enhanced log.
notice how every category path was fetched, for example https://www.torlock.com/images/torrents/black-jesus.html?sort=added&order=desc
https://www.torlock.com/audiobooks/torrents/black-jesus.html?sort=added&order=desc
etc.
but these paths should have been skipped as we were expecting only the TV category path to be used.
the categorymappings clause in the path block is supposed to only allow the path to be used if the category was in the filtered search list, or if the filter was not specified, i.e. all categories to be searched. (see wiki https://github.com/Jackett/Jackett/wiki/Definition-format, search block , categorymappings descriptions and examples).

**Comment by alecsg77** [comment]
Hi @garfield69, could you please confirm that the problem is still present in the current version (v0.17.963)?
I checked in version v0.11.802, and I could reproduce it, but I can't with the latest one.

Thanks

**Comment by ilike2burnthing** [comment]
The issue was resolved in the commit just above https://github.com/Jackett/Jackett/issues/6256#issuecomment-546726557 (i.e. https://github.com/Jackett/Jackett/commit/705038369ea57f8dd59464cded3c71ed9260fca6) by removing the additional search paths altogether.

To get it to work you'd have to use a slightly altered version of the example given in https://github.com/Jackett/Jackett/issues/6256#issuecomment-583775454, however I just tried it and it won't even accept it anymore, Jackett returns:
```
Exception during deserialization
 ---> System.Runtime.Serialization.SerializationException: Property 'categorymappings' not found on type 'Jackett.Common.Models.searchPathBlock'.
```
basically `categorymappings` can't be used for the search path anymore (presumably this got removed when the categorymappings code was redone last year).

The issue is still open in case someone is able to add the ability for this in the future.
